horovod/common/logging.cc

```diff
@@ -0,0 +1,91 @@
+#include <chrono>
+#include <algorithm>
+
+#include "logging.h"
+
+namespace horovod {
+namespace common {
+
+LogMessage::LogMessage(const char* fname, int line, LogLevel severity)
+    : fname_(fname), line_(line), severity_(severity) {}
+
+void LogMessage::GenerateLogMessage(bool log_time) {
+  if (log_time) {
+    auto now = std::chrono::system_clock::now();
+    auto as_time_t = std::chrono::system_clock::to_time_t(now);
+
+    auto duration = now.time_since_epoch();
+    auto seconds = std::chrono::duration_cast<std::chrono::seconds>(duration);
+    auto micros_remainder = std::chrono::duration_cast<std::chrono::microseconds>(duration - seconds);
+
+    const size_t time_buffer_size = 30;
+    char time_buffer[time_buffer_size];
+    strftime(time_buffer, time_buffer_size, "%Y-%m-%d %H:%M:%S",
+             localtime(&as_time_t));
+
+    fprintf(stdout, "[%s.%06d: %c %s:%d] %s\n", time_buffer, micros_remainder,
```

Comment:
Use `std::cout`?  It supports `%06d` formatting and etc.

Reply:
As a bonus, we could use `std::cout` for TRACE, DEBUG and INFO, and `std::cerr` for WARNING, ERROR and FATAL.

Reply:
On it